File: Selenium/Activities/activity17.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class activity17 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://www.training-support.net/selenium/selects");
        System.out.println("The title od the page is: " +driver.getTitle());
        //On the Single Select:
        WebElement dropdown = driver.findElement(By.id("single-select"));
        Select singleSelect = new Select(dropdown);
        //Select the second option using the visible text.
        singleSelect.selectByVisibleText("Option 2");
        System.out.println("Second option: " + singleSelect.getFirstSelectedOption().getText());

        //Select the third option using the index.
        singleSelect.selectByIndex(3);
        System.out.println("Third option: " + singleSelect.getFirstSelectedOption().getText());

        //Select the fourth option using the value.
        singleSelect.selectByValue("4");
        System.out.println("Fourth option: " + singleSelect.getFirstSelectedOption().getText());

        //Get all the options and print them to the console.
        List<WebElement> allOptions = singleSelect.getOptions();
        System.out.println("Options in the dropdown: ");
        for(WebElement option: allOptions){
            System.out.println(option.getText());
        }
        driver.quit();

    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.select import Select
from selenium.webdriver.common.by import By

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://www.training-support.net/selenium/selects")
    print("The title of the page is: ", driver.title)
    #On the Single Select:
    singleSelect = driver.find_element(By.ID, "single-select")
    dropdown = Select(singleSelect) 
    #Select the second option using the visible text.
    dropdown.select_by_visible_text("Option 2")
    print("The second option is: ", dropdown.first_selected_option.text)
    #Select the third option using the index.
    dropdown.select_by_index(3)
    print("The Third option is: ", dropdown.first_selected_option.text)
    #Select the fourth option using the value.
    dropdown.select_by_value("4")
    print("The Fourth option is: ", dropdown.first_selected_option.text)
    #Get all the options and print them to the console.
    select_options = dropdown.options
    for option in select_options:
        print("Option: ", option.text)